lib/modules/manager/pipenv/extract.ts

```diff
@@ -111,6 +105,21 @@ function extractFromSection(
   return deps;
 }
 
+function isPipRequirements(
+  section?:
+    | Record<string, PipRequirement>
+    | Record<string, string>
+    | PipSource[],
+): section is Record<string, PipRequirement> {
+  return (
+    !is.array(section) &&
+    is.object(section) &&
+    Object.entries(section).findIndex(
+      ([, dep]) => !is.object(dep) && !is.string('string'),
+    ) === -1
```

Comment:
use find method 
```suggestion
    Object.entries(section).find(
      (dep) => !is.object(dep),
    ) !== undefined 
```
`!is.string('string')` is always false

Reply:
`'string'` was supposed to be `dep`, not sure how that got by my review. I've fixed that and applied the other relevant parts of your suggestion.